src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,339 @@
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonDomainName);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonPassword);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonCount);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogon);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReason);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReasonUI);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeCaption);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeText);
+
+            //user specific            
+            if (!_windowsServiceHelper.HasActiveSession(domainName, userName))
+            {
+                string securityId = _windowsServiceHelper.GetSecurityId(domainName, userName);
+                RevertRegistrySettingsForDifferentUser(securityId);
+            }
+            else
+            {
+                RevertRegistrySettingsForCurrentUser();
+            }
+        }
+
+        public bool DoesRegistryExistForUser(string domainName, string userName)
+        {
+            if (!_windowsServiceHelper.HasActiveSession(domainName, userName))
+            {
+                string securityId = _windowsServiceHelper.GetSecurityId(domainName, userName);
+                return _registryManager.RegsitryExists(securityId);
+            }
+            return true;
+        }
+
+        private List<string> GetWarningsForUserSpecificSettings(string domainName, string userName)
+        {
+            string screenSaverValue = null;
+            List<string> warningReasons = new List<string>();
+
+            if (!_windowsServiceHelper.HasActiveSession(domainName, userName))
+            {
+                string securityId = _windowsServiceHelper.GetSecurityId(domainName, userName);
+                screenSaverValue = _registryManager.GetValue(RegistryHive.Users, $"{securityId}\\{RegistryConstants.SubKeys.ScreenSaverDomainPolicy}", RegistryConstants.ValueNames.ScreenSaver);
+            }
+            else
+            {
+                screenSaverValue = _registryManager.GetValue(RegistryHive.CurrentUser, RegistryConstants.SubKeys.ScreenSaverDomainPolicy, RegistryConstants.ValueNames.ScreenSaver);
+            }
+
+            if (int.TryParse(screenSaverValue, out int isScreenSaverDomainPolicySet)
+                    && isScreenSaverDomainPolicySet == 1)
+            {
+                warningReasons.Add(StringUtil.Loc("UITestingWarning_ScreenSaver"));
```

Comment:
I wonder if it will help during investigations if similar warnings are tested on agent startup (when the agent is launched via auto-logon) and dump to the diag log.